source/common/http/async_client_impl.cc

```diff
@@ -168,6 +168,7 @@ void AsyncStreamImpl::sendLocalReply(Code code, absl::string_view body,
                                      std::function<void(ResponseHeaderMap& headers)> modify_headers,
                                      const absl::optional<Grpc::Status::GrpcStatus> grpc_status,
                                      absl::string_view details) {
+  stream_info_.setResponseCodeDetails(details);
```

Comment:
Should we check the responseCodeDetails() not set first before setting it?

Reply:
sendLocalReply could happen after a responseDetails is set, I think it's fine to override here, as this "details" is the most update to date "details".

Reply:
No, because I want this to override whatever has already been set.